coopr-docs/docs/source/guide/superadmin/docker-automator-plugin.rst

```diff
@@ -0,0 +1,119 @@
+and to manage containers of these images.
+
+.. note:: The Docker Automator currently requires that Docker is installed, configured, and running on the target host(s), such as provided by CoreOS.
+
+
+As an example, consider the following Coopr service definition:
+::
+
+    {
+        "name": "cdap-standalone",
+        "description": "CDAP SDK (Standalone) running in a Docker container (caskdata/cdap-standalone)",
+        "provisioner": {
+            "actions": {
+                "install": {
+                    "type": "docker",
+                    "fields": {
+                        "image_name": "caskdata/cdap-standalone",
+                        "publish_ports": "9999,10000"
+                    }
+                },
+                "start": {
+                    "type": "docker",
+                    "fields": {
+                        "image_name": "caskdata/cdap-standalone",
+                        "publish_ports": "9999,10000"
+                    }
+                },
+                "stop": {
+                    "type": "docker",
+                    "fields": {
+                        "image_name": "caskdata/cdap-standalone",
+                        "publish_ports": "9999,10000"
+                    }
+                }
+            }
+        }
+    }
+
+This defines a service named "cdap-standalone" which has three defined actions, install, start, and stop.  When each
+action is invoked for this service, the ``type`` field indicates to the provisioner to use the Docker Automator plugin
+to manage the action.  The Docker Automator defines two custom fields: a required ``image_name`` and optional
+``publish_ports``.  The ``image_name`` field is the image or repository to pull from Docker Hub.  Note that the image
+must be a public image, hosted on Docker Hub.  The ``publish_ports`` field is optional, and is a comma-separated list
+of exposed container ports to publish on the Docker host.
+
+In this example, when the install action is invoked for this service, the Docker Automator will connect to the remote
+host(s), then effectively invoke the following command on them:
+:: 
+
+        docker pull caskdata/cdap-standalone
+
+When the start action is invoked for this service, the Docker Automator will connect to the remote host(s), then effectively
+invoke the following command on them:
+::
+
+        docker run -d -p9999:9999 -p10000:10000 caskdata/cdap-standalone
+
+Note that this example is a real service included with Coopr, which can be used to launch a container with the Cask Data Application
+Platform (CDAP) SDK services running!
+
+How it Works
+============
+
+A more in-depth look at Docker Automator:
+        1. During the "bootstrap" action:
+                a. The plugin verifies that ``docker`` is available on the host(s)
+        2. When a defined docker service action runs (ie, install, start, stop, etc):
```

Comment:
(for example: install, start, or stop): \[I try to avoid latin and abbreviations]